src/library/scala/collection/Seq.scala

```diff
@@ -774,11 +774,28 @@ trait SeqOps[+A, +CC[_], +C] extends Any
     *        x >  0       if this.length >  len
     *   }}}
     *  The method as implemented here does not call `length` directly; its running time
-    *  is `O(length min len)` instead of `O(length)`. The method should be overwritten
-    *  if computing `length` is cheap.
+    *  is `O(length min len)` instead of `O(length)`. The method should be overridden
+    *  if computing `length` is cheap and `knownSize` returns `-1`.
+    *
+    *  @see [[lengthIs]]
     */
   def lengthCompare(len: Int): Int = super.sizeCompare(len)
 
+  /** Compares the length of this $coll to the size of another `Iterable`.
+    *
+    *   @param   that the `Iterable` whose size is compared with this $coll's length.
+    *   @return  A value `x` where
+    *   {{{
+    *        x <  0       if this.length <  that.size
+    *        x == 0       if this.length == that.size
+    *        x >  0       if this.length >  that.size
+    *   }}}
+    *  The method as implemented here does not call `length` or `size` directly; its running time
+    *  is `O(this.length min that.size)` instead of `O(this.length + that.size)`.
+    *  The method should be overridden if computing `size` is cheap and `knownSize` returns `-1`.
```

Comment:
This comment is a bit misleading because the method is final.

I'd personally do the aliasing the other way around (`final def sizeCompare = lengthCompare`) so subclasses of `Seq` consistently have to override `length...` methods, and all `size...` methods are final. (I saw [@Ichoran's comment](https://github.com/scala/bug/issues/11388#issuecomment-470714127), but it didn't convince me 😄)

Reply:
oops, I missed that comment. I can reverse it, which will make that comment meaningful again lol

Reply:
I agree with @lrytz that there is an advantage of the regularity of having `size`everything final so you need to override `length`whatever.  For something that is just an alias, I will note that this is a bad idea in general because traits that extend `Iterable` but not `Seq` cannot override size calculations in cakes.  But cakes are out of fashion anyway, and we're already stuck with `length` vs `size` I guess; so may as well go all in and make all the `sizeX` defer to `lengthX`.

Reply:
done